students/TimPauley/session01/test_unit.py

```diff
@@ -0,0 +1,333 @@
+        item_dict = {}
+        item_dict["item_code"] = 2
+        item_dict["item_description"] = "b"
+        item_dict["item_rental_price"] = 2.0
+        item_dict["item_price"] = 2.0
+        item_dict["item_material"] = "a"
+        item_dict["size"] = "S"
+        f = Furniture(**item_dict)
+        new_dict = f.return_as_dictionary()
+        self.assertEqual(f.product_code, new_dict["product_code"])
+        self.assertEqual(f.description, new_dict["description"])
+        self.assertEqual(f.rental_price, new_dict["rental_price"])
+        self.assertEqual(f.material, new_dict["material"])
+        self.assertEqual(f.size, new_dict["size"])
+
+    def test_check_for_big_screen(self):
+        item_dict = {}
+        item_dict["item_code"] = 2
+        item_dict["item_description"] = "b"
+        item_dict["item_rental_price"] = 2.0
+        item_dict["item_price"] = 2.0
+        item_dict["item_material"] = "a"
+        item_dict["size"] = "S"
+        f = Furniture(**item_dict)
+        self.assertEqual(f.check_for_big_screen(), False)
+        f.size = "XL"
+        self.assertEqual(f.check_for_big_screen(), True)
+
+
+class Test_InventoryClass(TestCase):
+    def test_init(self):
+        item_dict = {}
+        item_dict["item_code"] = 2
+        item_dict["item_description"] = "b"
+        item_dict["item_rental_price"] = 2.0
+        item_dict["item_price"] = 2.0
+        i = Inventory(**item_dict)
+        self.assertEqual(i.product_code, 2)
+        self.assertEqual(i.description, "b")
+        self.assertEqual(i.rental_price, 2.0)
+        self.assertEqual(i.market_price, 2.0)
+
+    def test_return_as_dictionary(self):
+        item_dict = {}
+        item_dict["item_code"] = 2
+        item_dict["item_description"] = "b"
+        item_dict["item_rental_price"] = 2.0
+        item_dict["item_price"] = 2.0
+        i = Inventory(**item_dict)
+        new_dict = i.return_as_dictionary()
+        self.assertEqual(i.product_code, new_dict["product_code"])
+        self.assertEqual(i.description, new_dict["description"])
+        self.assertEqual(i.rental_price, new_dict["rental_price"])
+
+    def test_check_market_price(self):
+        item_dict = {}
+          item_dict["item_code"] = 2
+        item_dict["item_description"] = "b"
+        item_dict["item_rental_price"] = 2.0
+        item_dict["item_price"] = 2.0
+        i = Inventory(**item_dict)
+        self.assertEqual(i.check_market_price(), False)
+        i.market_price = get_latest_price()
+        self.assertEqual(i.check_market_price(), True)
+
+
+class Test_Main(TestCase):
+    def test_generate_menu(self):
+        self.assertEqual(generate_menu("Z"), False)
+        self.assertEqual(generate_menu("1").__name__, "add_new_item")
+        self.assertEqual(generate_menu("2").__name__, "item_info")
+        self.assertEqual(generate_menu("q").__name__, "exit_program")
+
+    def test_get_price(self):
+        self.assertEqual(get_price(), get_latest_price())
+
```

Comment:
I take it these tests aren't working. I didn't see a return_item_info function in your main. I'm guessing your plan is to write another version of get_info that returns instead of prints? You could just add a return statement to get_info that returns the same thing it is printing.